autosklearn/ensembles/multiobjective_dummy_ensemble.py

```diff
@@ -57,24 +59,21 @@ def __init__(
             self.metric = [metrics]
         self.random_state = random_state
         self.backend = backend
+        self.pareto_set_: list[SingleModelEnsemble] | None = None
```

Comment:
Hey, is there a reason you added this here? Doesn't this indicate to sklearn that this object has been fitted and therefore it should only be added at fit time?

Reply:
So the main reason is for the checkers. In general, setting attributes that are not in `__init__` will lead correctly to mypy error when trying to access it elsewhere as it may not have been set. I was listing out several options but the only one that makes sense is to do:

* Use the `@property pareto_set` everywhere internally and have it do the check it's doing along with `getattr(self, "pareto_set_"). This will raise with a nice error or return the attribute if it exists.

Reply:
Done.